drivers/rtc/rtc_max31329.c

```diff
@@ -0,0 +1,570 @@
+	regs[0] = 0x00;
+	ret = i2c_reg_write_byte_dt(&config->i2c, MAX31329_REG_RTC_CONFIG2, regs[0]);
+	if (ret) {
+		LOG_ERR("Error when setting intB: %i", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
+static int max31329_alarm_get_time(const struct device *dev, uint16_t id, uint16_t *mask,
+				   struct rtc_time *timeptr)
+{
+	const struct max31329_config *config = dev->config;
+	int ret = 0;
+	uint8_t raw_time[7] = {0};
+
+	if (id != 0) {
+		LOG_ERR(" ID %d not supporting", id);
+		return -ENOTSUP;
+	}
+
+	/*starting from seconds register read till year*/
+	ret = i2c_burst_read_dt(&config->i2c, MAX31329_REG_ALM1_SEC, raw_time, sizeof(raw_time));
+	if (ret) {
+		LOG_ERR("Fail to get time (error %d)", ret);
+		return ret;
+	}
+
+	timeptr->tm_nsec = 0;
+	/* Get seconds */
+	timeptr->tm_sec = bcd2bin(raw_time[0]);
+	/* Get minutes */
+	timeptr->tm_min = bcd2bin(raw_time[1]);
+	/* Get hours in 24hr format */
+	timeptr->tm_hour = bcd2bin(raw_time[2] & MAX31329_HOURS_MASK);
+	/* Get days */
+	timeptr->tm_wday = bcd2bin(raw_time[3] & MAX31329_WEEKDAYS_MASK);
+	/* Get weekdays */
+	timeptr->tm_mday = bcd2bin(raw_time[4] & MAX31329_DAYS_MASK);
+	/* Get month */
+	timeptr->tm_mon = bcd2bin(raw_time[5] & MAX31329_MONTHS_MASK);
+	/* we need 0 based(0-11)*/
+	timeptr->tm_mon -= 1;
+	/* Get year */
+	/*Century Bit is ignored */
+	timeptr->tm_year = bcd2bin(raw_time[6]);
+	if (IS_BIT_SET(raw_time[5], 7)) {
+		timeptr->tm_year += 100;
+	}
+	return 0;
+	LOG_ERR("Not Supported yet");
+	return -EINVAL;
+}
+
+static int max31329_alarm_is_pending(const struct device *dev, uint16_t id)
+{
+	const struct max31329_config *config = dev->config;
+	uint8_t status;
+	int ret;
+
+	/* Read the status register from the RTC*/
+	ret = i2c_reg_read_byte_dt(&config->i2c, MAX31329_REG_STATUS, &status);
+	if (ret) {
+		LOG_ERR("Error when reading status: %i", ret);
+		return ret;
+	}
+
+	/* Check the alarm interrupt flag*/
+	return (status & BIT(id ? 1 : 0)) != 0;
+}
+#endif
+
+#ifdef MAX31329_INT_GPIOS_IN_USE
+void callback_work_handler(struct k_work *work)
+{
+	/* This function is run as a work so the user can spend here all the necessary time */
+	struct max31329_data *data = CONTAINER_OF(work, struct max31329_data, callback_work);
+
+	if (data->alarm_callback == NULL) {
+		LOG_WRN("No MAX31329 alarm callback function provided");
+	} else {
+		data->alarm_callback(data->dev, 0, data->alarm_user_data);
+	}
+}
+
+/* The function called when the clock alarm activates the interrupt*/
+void gpio_callback_function(const struct device *dev, struct gpio_callback *cb, uint32_t pins)
+{
+	struct max31329_data *data = CONTAINER_OF(cb, struct max31329_data, intb_callback);
+
+	LOG_DBG("MAX31329 interrupt detected");
+	/* By using a work we are able to run "heavier" code */
+	k_work_submit(&(data->callback_work));
+}
+#endif
+
+static int max31329_alarm_set_callback(const struct device *dev, uint16_t id,
+				       rtc_alarm_callback callback, void *user_data)
```

Comment:
To disable the callback, the `callback` arg can be NULL, this implementation does not support disabling the interrupt from what I see :)